src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,216 @@
+package duke.parser;
+
+import java.util.List;
+
+import duke.command.AddTaskCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.data.DateAndTime;
+import duke.data.exceptions.DukeException;
+import duke.data.exceptions.EmptyTaskDescriptionException;
+import duke.data.exceptions.InvalidDateAndTimeException;
+import duke.data.exceptions.InvalidInputException;
+
+/**
+ * Represents a parser that makes sense of what the user input is.
+ */
+public class Parser {
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+    private static final String EXIT = "bye";
+    private static final String LIST = "list";
+    private static final String DONE = "done";
+    private static final String DELETE = "delete";
+    private static final String FIND = "find";
```

Comment:
Perhaps it would be better to implement these as enums?

Reply:
Ok thank you for reviewing :)!